portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -41,10 +44,34 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String content = request.getParameter("comment");
     String name = request.getParameter("name");
-    if (content.length() != 0) {
+    
+    if (content != null && name != null && content.length() != 0) {
       Comment comment = name.length() == 0 ? new Comment(content) : new Comment(name, content);
+      Entity commentEntity = new Entity("Comment");
+      commentEntity.setProperty("name", comment.getNAME());
+      commentEntity.setProperty("content", comment.getCONTENT());
+      commentEntity.setProperty("numLikes", comment.getNumLikes());
+      commentEntity.setProperty("timestamp", comment.getTIMESTAMP());
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      datastore.put(commentEntity);
       commentsList.add(comment);
+    } else {
```

Comment:
so we get into this else block if any of the if conditions (content != null && name != null && content.length() != 0) 
are not met, do any of those conditions not being met signify that this is a comment is liked case? 
(just checking)

Reply:
I changed this in my next commit (d3ae78f) but still am not sure if it's clear.. I'll leave a comment about this in that commit